torch/package/package_exporter.py

```diff
@@ -468,8 +492,23 @@ def add_dependency(self, module_name: str, dependencies=True):
 
                 # If we are interning this module, we need to retrieve its
                 # dependencies and package those as well.
-                if pattern_info.action == _ModuleProviderAction.INTERN:
+                if action == _ModuleProviderAction.INTERN:
                     self._intern_module(module_name, dependencies)
+                if action == _ModuleProviderAction.SELECTIVE_EXTERN:
+                    module_obj = self._import_module(module_name)
+                    is_package = hasattr(module_obj, "__path__")
+                    source = self._get_source_of_module(module_obj)
```

Comment:
We don't really need `is_package` or `source`. The source is discarded since we are replacing it with our shim, and `is_package` must be true for selective intern to even make sense.

Reply:
Yeah with source I think that's correct, the is_package is for the assert to make sure we aren't doing anything obviously wrong.